### PR TITLE
Fix building TRex Simulator (./linux) and more

### PR DESCRIPTION
**Pull request by mateumann**
And also some minor changes:
* rebase onto Artur's changes
* enable PTP receiving part (a.k.a. PTP slave) handled by RXTimesync class
* rename timestamping methods in CGlobalInfo::m_options
* fix the typo in RXTimesync::advertise() method

### CODE REVIEW

**Review by mkczuk — APPROVED** [review]

File: linux/ws_main.py [hunk 1]
```diff
@@ -227,6 +227,7 @@ def configure_sanitized (conf):
             'bp_sim_tcp.cpp',
             'inet_pton.cpp',
             'trex_global.cpp',
+             'trex_timesync.cpp',
```
Comment (MehTheHedgehog) [inline linux/ws_main.py]: one space to much?

File: src/common/Network/Packet/PTPPacket.cpp [hunk 2]
```diff
@@ -18,32 +18,38 @@ limitations under the License.
 
 void PTP::Header::dump(FILE *fd) {
     fprintf(fd, "\nPTP Header");
-    fprintf(fd, "\nMessageId 0x%.1X (%s), TransportSpecific 0x%.1X, PTP Version %d, MessageLength %d", trn_and_msg.msg_type_raw(),
-            trn_and_msg.msg_type_str(), trn_and_msg.trans_spec_raw(), *ver, *message_len);
+    fprintf(fd, "\nMessageId 0x%.1X (%s), TransportSpecific 0x%.1X, PTP Version %d, MessageLength %d",
+            trn_and_msg.msg_type_raw(), trn_and_msg.msg_type_str(), trn_and_msg.trans_spec_raw(), (int)*ver,
+            (int)*message_len);
```
Comment (MehTheHedgehog) [inline src/common/Network/Packet/PTPPacket.cpp]: why cast everything to int?
static_cast, btw

File: src/trex_timesync.cpp [hunk 3]
```diff
@@ -119,3 +117,7 @@ const char *CTimesyncEngine::descTimesyncState(int port) {
     }
     return "NONE";
 }
+
+timespec CTimesyncEngine::timestampToTimespec(uint64_t timestamp) {
+    return {(uint32_t)(timestamp / (1000 * 1000 * 1000)), (uint32_t)(timestamp % (1000 * 1000 * 1000))};
+};
```
Comment (MehTheHedgehog) [inline src/trex_timesync.cpp]: you can add nl

File: src/common/Network/Packet/PTPPacket.cpp [hunk 4]
```diff
@@ -18,32 +18,38 @@ limitations under the License.
...
+    fprintf(fd, "\nSeconds (msb) 0x%.4X (%u), Seconds (lsb) (0x%.8lX (%lu)), Nanoseconds (0x%.8lX (%lu))",
+            (unsigned int)*(origin_timestamp.sec_msb), (unsigned int)*(origin_timestamp.sec_msb),
+            (unsigned long int)*(origin_timestamp.sec_lsb), (unsigned long int)*(origin_timestamp.sec_lsb),
+            (unsigned long int)*(origin_timestamp.ns), (unsigned long int)*(origin_timestamp.ns));
+    fprintf(fd, "\nTimestamp %ld.%ld", (unsigned long int)origin_timestamp.get_timestamp().tv_sec,
+            (unsigned long int)origin_timestamp.get_timestamp().tv_nsec);
+    fprintf(fd, "\nRequesting ClockIdentity 0x%.16lX, Requesting SourcePortId %d",
+            (unsigned long int)req_clock_identity.clock_id(), (unsigned int)req_clock_identity.port_number());
     fprintf(fd, "\n");
 }
```
Comment (MehTheHedgehog) [inline src/common/Network/Packet/PTPPacket.cpp]: you can add nl

**Review by MehTheHedgehog — CHANGES_REQUESTED** [review]

File: src/common/Network/Packet/PTPPacket.cpp (hunk 2 above)
Comment (mateumann) [reply src/common/Network/Packet/PTPPacket.cpp]: compiler threw errors when not casted

File: src/trex_timesync.cpp (hunk 3 above)
Comment (mateumann) [reply src/trex_timesync.cpp]: auto-formatting...

File: src/common/Network/Packet/PTPPacket.cpp (hunk 4 above)
Comment (mateumann) [reply src/common/Network/Packet/PTPPacket.cpp]: ditto

File: src/common/Network/Packet/PTPPacket.cpp (hunk 2 above)
Comment (mateumann) [reply src/common/Network/Packet/PTPPacket.cpp]: switched to `static_cast<>`

File: src/trex_timesync.cpp (hunk 3 above)
Comment (MehTheHedgehog) [reply src/trex_timesync.cpp]: github is angry because no newline at the eno of file

Comment (mateumann) [reply src/trex_timesync.cpp]: I have fed him.  Should be calmer now.

**Review by MehTheHedgehog — APPROVED** [review]

**Pull request merged**